hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/DeleteKeyHandler.java

```diff
@@ -35,16 +48,117 @@
     description = "deletes an existing key")
 public class DeleteKeyHandler extends KeyHandler {
 
+  @CommandLine.Option(names = "--skipTrash",
+      description = "Specify whether to skip Trash ")
+  private boolean skipTrash = false;
+
+  private static final Path CURRENT = new Path("Current");
+
   @Override
   protected void execute(OzoneClient client, OzoneAddress address)
       throws IOException, OzoneClientException {
 
     String volumeName = address.getVolumeName();
     String bucketName = address.getBucketName();
-    String keyName = address.getKeyName();
-
     OzoneVolume vol = client.getObjectStore().getVolume(volumeName);
     OzoneBucket bucket = vol.getBucket(bucketName);
-    bucket.deleteKey(keyName);
+    String keyName = address.getKeyName();
+
+    if (bucket.getBucketLayout().isFileSystemOptimized()) {
+      // Handle FSO delete key which supports trash also
+      deleteFSOKey(bucket, keyName);
+    } else {
+      bucket.deleteKey(keyName);
+    }
+  }
+
+  private void deleteFSOKey(OzoneBucket bucket, String keyName)
+      throws IOException {
+    float hadoopTrashInterval = getConf().getFloat(
+        FS_TRASH_INTERVAL_KEY, FS_TRASH_INTERVAL_DEFAULT);
+    long trashInterval =
+        (long) (getConf().getFloat(
+            OMConfigKeys.OZONE_FS_TRASH_INTERVAL_KEY,
+            hadoopTrashInterval) * 10000);
+
+    // If Bucket layout is FSO and Trash is enabled
+    // In this case during delete operation move key to trash
+    if (trashInterval > 0 && !skipTrash &&
```

Comment:
Since we are adding the skipTrash flag here, just a minor suggestion to reduce one operation in if condition for trashInterval and  check like below:
if (!skipTrash && trashInterval > 0  && !keyName.contains(TRASH_PREFIX)) as skipTrash itself may be false.

Reply:
trashInterval is used to determine whether trash is enabled or not, So I think first we should check whether trash enabled or not then we can check trash related configs.